src/garage/experiment/experiment.py

```diff
@@ -318,3 +201,279 @@ def to_local_command(params,
+            return self.function(ctxt, *args, **kwargs)
+
+
+def wrap_experiment(function=None,
+                    *,
+                    log_dir=None,
+                    prefix='experiment',
+                    name=None,
+                    snapshot_mode='last',
+                    snapshot_gap=1):
+    """Decorate a function to turn it into an ExperimentTemplate.
+
+    When invoked, the wrapped function will receive an ExperimentContext, which
+    will contain the log directory into which the experiment should log
+    information.
+
+    This decorator can be invoked in two differed ways.
+
+    Without arguments, like this:
+
+        @wrap_experiment
+        def my_experiment(ctxt, seed, lr=0.5):
+            ...
+
+    Or with arguments:
+
+        @wrap_experiment(snapshot_mode='all')
+        def my_experiment(ctxt, seed, lr=0.5):
+            ...
+
+    All arguments must be keyword arguments.
+
+    Args:
+        function (callable or None): The experiment function to wrap.
+        log_dir (str or None): The full log directory to log to. Will be
+            computed from `name` if omitted.
+        name (str or None): The name of this experiment template. Will be
+            filled from the wrapped function's name if omitted.
+        prefix (str): Directory under data/local in which to place the
+            experiment directory.
+        snapshot_mode (str): Policy for which snapshots to keep (or make at
+            all). Can be either "all" (all iterations will be saved), "last"
+            (only the last iteration will be saved), "gap" (every snapshot_gap
+            iterations are saved), or "none" (do not save snapshots).
+        snapshot_gap (int): Gap between snapshot iterations. Waits this number
+            of iterations before taking another snapshot.
+
+    Returns:
+        callable: The wrapped function.
+
+    """
+    return ExperimentTemplate(function=function,
+                              log_dir=log_dir,
+                              prefix=prefix,
+                              name=name,
+                              snapshot_mode=snapshot_mode,
+                              snapshot_gap=snapshot_gap)
+
+
+def dump_json(filename, data):
+    """Dump a dictionary to a file in JSON format.]
+
+    Args:
+        filename(str): Filename for the file.
+        data(dict): Data to save to file.
+
+    """
+    pathlib.Path(os.path.dirname(filename)).mkdir(parents=True, exist_ok=True)
+    with open(filename, 'w') as f:
+        json.dump(data, f, indent=2, sort_keys=True, cls=LogEncoder)
+
+
+class LogEncoder(json.JSONEncoder):
+    """Encoder to be used as cls in json.dump."""
+
+    def default(self, o):
+        """Perform JSON encoding.
+
+        Args:
+            o (object): Object to encode.
+
+        Returns:
+            str: Object encoded in JSON.
+
+        """
+
+        # Why is this method hidden? What does that mean?
+        # pylint: disable=method-hidden
```

Comment:
i think it means that this method is overriding some symbol which is not its same kind (maybe a classmethod or property?)

Reply:
Maybe. I didn't actually modify this code, I just moved it from `experiment_wrapper.py`.

Reply:
It's completely possible that this code doesn't actually work. Certainly no one has written the corresponding parsing code.